src/syscalls/mod.rs

```diff
@@ -345,8 +347,152 @@ pub unsafe extern "C" fn sys_open(name: *const c_char, flags: i32, mode: u32) ->
 	}
 }
 
+#[hermit_macro::system]
+#[unsafe(no_mangle)]
+pub unsafe extern "C" fn sys_getcwd(buf: *mut c_char, size: usize) -> *const c_char {
+	let error = |e: Errno| {
+		e.set_errno();
+		null::<c_char>()
+	};
+
+	if size == 0 {
+		return error(Errno::Inval);
+	}
+
+	let cwd = fs::get_cwd();
+	if let Err(e) = cwd {
+		return error(e);
+	}
+
+	let Ok(cwd) = cwd else { unreachable!() };
```

Comment:
We can use `match` here:

```suggestion
    let cwd = match cwd {
        Ok(cwd) => cwd,
        Err(err) => return -i32::from(err),
    };
```